app/src/main/java/com/aptoide/uploader/apps/InstallManager.kt

```diff
@@ -0,0 +1,48 @@
+package com.aptoide.uploader.apps
+
+import android.util.Log
+import com.flurry.sdk.t
+import io.reactivex.Completable
+
+class InstallManager(private val installedRepository: InstalledRepository,
+                     private val packageManagerInstalledAppsProvider: PackageManagerInstalledAppsProvider) {
+
+  fun insertAllInstalled(): Completable {
+    return packageManagerInstalledAppsProvider.installedApps
+        .doOnError{throwable -> Log.e("APP-85","Error "+ throwable.printStackTrace())}
+        .flatMapCompletable { installed ->
+          Log.d("APP-85", "insertAllInstalled: installedApps size "+  installed.size)
+          installedRepository.replaceAllBy(installed)
+        }
+  }
+
```

Comment:
Yes, it will replace the list everytime it runs (every time the app is open closed and opened again). I will change it to run only once after installation